README.md

```diff
@@ -1,2 +1,39 @@
-# anyc_msg_handling
-POC project to demonstrate new capabilities of C++17 in networking.
+# Asynchronous message handling
+This project aims to prove the power of the modern C++ (17) capabilities in
+writing high-performance networking code to be used later in 5G gNB networking
+services.
+
+## Dependencies
+1. cmake;
+2. boost.asio.
+
+## Solution architecture
+
+                      ┌────────────────┐
+                      │          perf  │
+                   ┌──┴───────────┐    ├───────────┐
+           TCP     │              │    │           │
+         ┌─────────┤ C++11 based  ├────┘           │
+         │         │  async app   │            ┌───▼────┐
+  ┌──────┴──────┐  └──────────────┘            │        │
+  │             │                              │ report │
+  │ TCP Traffic │                              │        │
+  │  Generator  │     ┌────────────────┐       └───▲────┘
+  │             │     │          perf  │           │
+  └──────┬──────┘  ┌──┴───────────┐    ├───────────┘
+         │         │              │    │
+         └─────────┤ c++17 based  ├────┘
+           TCP     │  async app   │
+                   └──────────────┘
+
+
+### Syntetic net protocol layout
+
+  ┌────────┬────────┬─────────────┬──────────────────────┐
+  │protocol│inverse │payload len. │                      │
```

Comment:
Maybe put payload message type here?

Reply:
protocol version replaced with the "Message Type" field